drivers/soundwire/stream.c

```diff
@@ -1663,6 +1673,11 @@ int sdw_deprepare_stream(struct sdw_stream_runtime *stream)
 
 	sdw_acquire_bus_lock(stream);
 
+	if (stream->state == SDW_STREAM_DEPREPARED) {
+		ret = 0;
+		goto state_err;
```

Comment:
state_err is not a good name.  can we  do as 
````c
sdw_release_bus_lock(stream);
return 0;
````

Reply:
state_err is what is already in the code, so might as well use it. I don't like duplication of unlock code, it's better a single flow for unlocks, much easier to track.

No change